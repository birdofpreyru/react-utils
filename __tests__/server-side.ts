import mockdate from 'mockdate';

mockdate.set('2020-04-19Z');

import type * as SrcM from '../src';

// eslint-disable-next-line @typescript-eslint/no-require-imports
const lib = require('../src') as typeof SrcM;

test('Export at server side', () => {
  expect(lib).toMatchSnapshot();
  expect(Object.keys(lib.time)).toMatchSnapshot();

  // The order of these entries in "serverExports" below must be enforced,
  // to keep existing snapshots oder.
  const ORDER: Record<string, number> = {
    SCRIPT_LOCATIONS: 1,
    getDefaultCspSettings: 2,
    errors: 3,
  };

  const serverExports = Object.entries(lib.server ?? {});
  serverExports.sort((a, b) => {
    const oA = ORDER[a[0]];
    const oB = ORDER[b[0]];
    if (oA && oB) return oA - oB;
    if (oA) return -1;
    if (oB) return 1;
    return a[0].localeCompare(b[0]);
  });

  serverExports.forEach((item) => expect(item).toMatchSnapshot());
});
