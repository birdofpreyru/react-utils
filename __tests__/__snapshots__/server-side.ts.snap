// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export at server side 1`] = `
{
  "Barrier": [Function],
  "BaseButton": [Function],
  "BaseModal": [Function],
  "Button": [Function],
  "Checkbox": [Function],
  "CustomDropdown": [Function],
  "Dropdown": [Function],
  "Emitter": [Function],
  "GlobalStateProvider": [Function],
  "Input": [Function],
  "Link": [Function],
  "MetaTags": [Function],
  "Modal": [Function],
  "NavLink": [Function],
  "PageLayout": [Function],
  "Semaphore": [Function],
  "Switch": [Function],
  "TextArea": [Function],
  "ThemeProvider": [Function],
  "Throbber": [Function],
  "WithTooltip": [Function],
  "YouTubeVideo": [Function],
  "client": undefined,
  "config": Config {
    "SECRET": {
      "dummySecretKey": "Dummy Secret Value",
    },
    "dummyKey": "Dummy Value",
  },
  "getGlobalState": [Function],
  "getSsrContext": [Function],
  "isomorphy": {
    "IS_CLIENT_SIDE": false,
    "IS_SERVER_SIDE": true,
    "buildTimestamp": [Function],
    "getBuildInfo": [Function],
    "isDevBuild": [Function],
    "isProdBuild": [Function],
  },
  "newAsyncDataEnvelope": [Function],
  "server": [Function],
  "splitComponent": [Function],
  "themed": [Function],
  "time": [Function],
  "useAsyncCollection": [Function],
  "useAsyncData": [Function],
  "useGlobalState": [Function],
  "webpack": {
    "requireWeak": [Function],
    "resolveWeak": [Function],
  },
  "withGlobalStateType": [Function],
  "withRetries": [Function],
}
`;

exports[`Export at server side 2`] = `
[
  "extend",
  "locale",
  "isDayjs",
  "unix",
  "en",
  "Ls",
  "p",
  "DAY_MS",
  "HOUR_MS",
  "MIN_MS",
  "SEC_MS",
  "YEAR_MS",
  "now",
  "timer",
  "useCurrent",
  "useTimezoneOffset",
]
`;

exports[`Export at server side 3`] = `
[
  "SCRIPT_LOCATIONS",
  {
    "BODY_OPEN": "BODY_OPEN",
    "DEFAULT": "DEFAULT",
    "HEAD_OPEN": "HEAD_OPEN",
  },
]
`;

exports[`Export at server side 4`] = `
[
  "getDefaultCspSettings",
  [Function],
]
`;

exports[`Export at server side 5`] = `
[
  "errors",
  {
    "CODES": {
      "100": "CONTINUE",
      "101": "SWITCHING_PROTOCOLS",
      "102": "PROCESSING",
      "103": "EARLY_HINTS",
      "200": "OK",
      "201": "CREATED",
      "202": "ACCEPTED",
      "203": "NON_AUTHORITATIVE_INFORMATION",
      "204": "NO_CONTENT",
      "205": "RESET_CONTENT",
      "206": "PARTIAL_CONTENT",
      "207": "MULTI_STATUS",
      "300": "MULTIPLE_CHOICES",
      "301": "MOVED_PERMANENTLY",
      "302": "MOVED_TEMPORARILY",
      "303": "SEE_OTHER",
      "304": "NOT_MODIFIED",
      "305": "USE_PROXY",
      "307": "TEMPORARY_REDIRECT",
      "308": "PERMANENT_REDIRECT",
      "400": "BAD_REQUEST",
      "401": "UNAUTHORIZED",
      "402": "PAYMENT_REQUIRED",
      "403": "FORBIDDEN",
      "404": "NOT_FOUND",
      "405": "METHOD_NOT_ALLOWED",
      "406": "NOT_ACCEPTABLE",
      "407": "PROXY_AUTHENTICATION_REQUIRED",
      "408": "REQUEST_TIMEOUT",
      "409": "CONFLICT",
      "410": "GONE",
      "411": "LENGTH_REQUIRED",
      "412": "PRECONDITION_FAILED",
      "413": "REQUEST_TOO_LONG",
      "414": "REQUEST_URI_TOO_LONG",
      "415": "UNSUPPORTED_MEDIA_TYPE",
      "416": "REQUESTED_RANGE_NOT_SATISFIABLE",
      "417": "EXPECTATION_FAILED",
      "418": "IM_A_TEAPOT",
      "419": "INSUFFICIENT_SPACE_ON_RESOURCE",
      "420": "METHOD_FAILURE",
      "421": "MISDIRECTED_REQUEST",
      "422": "UNPROCESSABLE_ENTITY",
      "423": "LOCKED",
      "424": "FAILED_DEPENDENCY",
      "426": "UPGRADE_REQUIRED",
      "428": "PRECONDITION_REQUIRED",
      "429": "TOO_MANY_REQUESTS",
      "431": "REQUEST_HEADER_FIELDS_TOO_LARGE",
      "451": "UNAVAILABLE_FOR_LEGAL_REASONS",
      "500": "INTERNAL_SERVER_ERROR",
      "501": "NOT_IMPLEMENTED",
      "502": "BAD_GATEWAY",
      "503": "SERVICE_UNAVAILABLE",
      "504": "GATEWAY_TIMEOUT",
      "505": "HTTP_VERSION_NOT_SUPPORTED",
      "507": "INSUFFICIENT_STORAGE",
      "511": "NETWORK_AUTHENTICATION_REQUIRED",
      "ACCEPTED": 202,
      "BAD_GATEWAY": 502,
      "BAD_REQUEST": 400,
      "CONFLICT": 409,
      "CONTINUE": 100,
      "CREATED": 201,
      "EARLY_HINTS": 103,
      "EXPECTATION_FAILED": 417,
      "FAILED_DEPENDENCY": 424,
      "FORBIDDEN": 403,
      "GATEWAY_TIMEOUT": 504,
      "GONE": 410,
      "HTTP_VERSION_NOT_SUPPORTED": 505,
      "IM_A_TEAPOT": 418,
      "INSUFFICIENT_SPACE_ON_RESOURCE": 419,
      "INSUFFICIENT_STORAGE": 507,
      "INTERNAL_SERVER_ERROR": 500,
      "LENGTH_REQUIRED": 411,
      "LOCKED": 423,
      "METHOD_FAILURE": 420,
      "METHOD_NOT_ALLOWED": 405,
      "MISDIRECTED_REQUEST": 421,
      "MOVED_PERMANENTLY": 301,
      "MOVED_TEMPORARILY": 302,
      "MULTIPLE_CHOICES": 300,
      "MULTI_STATUS": 207,
      "NETWORK_AUTHENTICATION_REQUIRED": 511,
      "NON_AUTHORITATIVE_INFORMATION": 203,
      "NOT_ACCEPTABLE": 406,
      "NOT_FOUND": 404,
      "NOT_IMPLEMENTED": 501,
      "NOT_MODIFIED": 304,
      "NO_CONTENT": 204,
      "OK": 200,
      "PARTIAL_CONTENT": 206,
      "PAYMENT_REQUIRED": 402,
      "PERMANENT_REDIRECT": 308,
      "PRECONDITION_FAILED": 412,
      "PRECONDITION_REQUIRED": 428,
      "PROCESSING": 102,
      "PROXY_AUTHENTICATION_REQUIRED": 407,
      "REQUESTED_RANGE_NOT_SATISFIABLE": 416,
      "REQUEST_HEADER_FIELDS_TOO_LARGE": 431,
      "REQUEST_TIMEOUT": 408,
      "REQUEST_TOO_LONG": 413,
      "REQUEST_URI_TOO_LONG": 414,
      "RESET_CONTENT": 205,
      "SEE_OTHER": 303,
      "SERVICE_UNAVAILABLE": 503,
      "SWITCHING_PROTOCOLS": 101,
      "TEMPORARY_REDIRECT": 307,
      "TOO_MANY_REQUESTS": 429,
      "UNAUTHORIZED": 401,
      "UNAVAILABLE_FOR_LEGAL_REASONS": 451,
      "UNPROCESSABLE_ENTITY": 422,
      "UNSUPPORTED_MEDIA_TYPE": 415,
      "UPGRADE_REQUIRED": 426,
      "USE_PROXY": 305,
    },
    "ERRORS": {
      "ACCEPTED": "Accepted",
      "BAD_GATEWAY": "Bad Gateway",
      "BAD_REQUEST": "Bad Request",
      "CONFLICT": "Conflict",
      "CONTINUE": "Continue",
      "CREATED": "Created",
      "EARLY_HINTS": "Early Hints",
      "EXPECTATION_FAILED": "Expectation Failed",
      "FAILED_DEPENDENCY": "Failed Dependency",
      "FORBIDDEN": "Forbidden",
      "GATEWAY_TIMEOUT": "Gateway Timeout",
      "GONE": "Gone",
      "HTTP_VERSION_NOT_SUPPORTED": "HTTP Version Not Supported",
      "IM_A_TEAPOT": "I'm a teapot",
      "INSUFFICIENT_SPACE_ON_RESOURCE": "Insufficient Space on Resource",
      "INSUFFICIENT_STORAGE": "Insufficient Storage",
      "INTERNAL_SERVER_ERROR": "Internal Server Error",
      "LENGTH_REQUIRED": "Length Required",
      "LOCKED": "Locked",
      "METHOD_FAILURE": "Method Failure",
      "METHOD_NOT_ALLOWED": "Method Not Allowed",
      "MISDIRECTED_REQUEST": "Misdirected Request",
      "MOVED_PERMANENTLY": "Moved Permanently",
      "MOVED_TEMPORARILY": "Moved Temporarily",
      "MULTIPLE_CHOICES": "Multiple Choices",
      "MULTI_STATUS": "Multi-Status",
      "NETWORK_AUTHENTICATION_REQUIRED": "Network Authentication Required",
      "NON_AUTHORITATIVE_INFORMATION": "Non Authoritative Information",
      "NOT_ACCEPTABLE": "Not Acceptable",
      "NOT_FOUND": "Not Found",
      "NOT_IMPLEMENTED": "Not Implemented",
      "NOT_MODIFIED": "Not Modified",
      "NO_CONTENT": "No Content",
      "OK": "OK",
      "PARTIAL_CONTENT": "Partial Content",
      "PAYMENT_REQUIRED": "Payment Required",
      "PERMANENT_REDIRECT": "Permanent Redirect",
      "PRECONDITION_FAILED": "Precondition Failed",
      "PRECONDITION_REQUIRED": "Precondition Required",
      "PROCESSING": "Processing",
      "PROXY_AUTHENTICATION_REQUIRED": "Proxy Authentication Required",
      "REQUESTED_RANGE_NOT_SATISFIABLE": "Requested Range Not Satisfiable",
      "REQUEST_HEADER_FIELDS_TOO_LARGE": "Request Header Fields Too Large",
      "REQUEST_TIMEOUT": "Request Timeout",
      "REQUEST_TOO_LONG": "Request Entity Too Large",
      "REQUEST_URI_TOO_LONG": "Request-URI Too Long",
      "RESET_CONTENT": "Reset Content",
      "SEE_OTHER": "See Other",
      "SERVICE_UNAVAILABLE": "Service Unavailable",
      "SWITCHING_PROTOCOLS": "Switching Protocols",
      "TEMPORARY_REDIRECT": "Temporary Redirect",
      "TOO_MANY_REQUESTS": "Too Many Requests",
      "UNAUTHORIZED": "Unauthorized",
      "UNAVAILABLE_FOR_LEGAL_REASONS": "Unavailable For Legal Reasons",
      "UNPROCESSABLE_ENTITY": "Unprocessable Entity",
      "UNSUPPORTED_MEDIA_TYPE": "Unsupported Media Type",
      "UPGRADE_REQUIRED": "Upgrade Required",
      "USE_PROXY": "Use Proxy",
    },
    "assert": [Function],
    "fail": [Function],
    "getErrorForCode": [Function],
    "joi": {
      "ValidationError": [Function],
      "_types": Set {
        "alternatives",
        "any",
        "array",
        "boolean",
        "date",
        "function",
        "link",
        "number",
        "object",
        "string",
        "symbol",
        "binary",
      },
      "allow": [Function],
      "alt": [Function],
      "alternatives": [Function],
      "any": [Function],
      "array": [Function],
      "assert": [Function],
      "attempt": [Function],
      "binary": [Function],
      "bool": [Function],
      "boolean": [Function],
      "build": [Function],
      "cache": {
        "provision": [Function],
      },
      "checkPreferences": [Function],
      "compile": [Function],
      "custom": [Function],
      "date": [Function],
      "defaults": [Function],
      "disallow": [Function],
      "equal": [Function],
      "exist": [Function],
      "expression": [Function],
      "extend": [Function],
      "forbidden": [Function],
      "func": [Function],
      "function": [Function],
      "in": [Function],
      "invalid": [Function],
      "isError": [Function],
      "isExpression": [Function],
      "isRef": [Function],
      "isSchema": [Function],
      "link": [Function],
      "not": [Function],
      "number": [Function],
      "object": [Function],
      "only": [Function],
      "optional": [Function],
      "options": [Function],
      "override": Symbol(override),
      "preferences": [Function],
      "prefs": [Function],
      "ref": [Function],
      "required": [Function],
      "string": [Function],
      "strip": [Function],
      "symbol": [Function],
      "trace": [Function],
      "types": [Function],
      "untrace": [Function],
      "valid": [Function],
      "version": "17.13.3",
      "when": [Function],
      "x": [Function],
      Symbol(@hapi/lab/coverage/initialize): [Function],
    },
    "newError": [Function],
  },
]
`;
